requirements.txt

```diff
@@ -2,17 +2,16 @@ altair>=4.0.0,<5
 Click>=7.1.2
 colorama>=0.4.3
 cryptography>=3.2
-dataclasses
 importlib-metadata>=1.7.0 # (included in Python 3.8 by default.)
 Ipython>=7.16.3
 jinja2>=2.10,<3.1.0 # contextfilter has been renamed
 jsonpatch>=1.22
 jsonschema>=2.5.1
 mistune>=0.8.4
 nbformat>=5.0
-numpy>=1.14.1
+numpy>=1.14.2
 packaging
-pandas>=0.23.0
+pandas>=0.24.0
```

Comment:
Should we be bumping these? I decided to here just to start a conversation - happy to revert these if needed but I don't have the full context around why these are our minimum versions.

Reply:
Not sure if we should change those for now... might be good to adjust after @anthonyburdi has the usage stats about packages in installed environments hooked up.

Even if we end up not bumping those, since `Python36-Pandas023` was removed from test matrix, can you cleanup the special test logic for old versions of pandas in this branch/PR too?

```
% grep -R -E '(pandas_022|pandas_023|pandas>=024)' .
./tests/expectations/test_util.py:        only_for=["pandas>=024"],
./tests/expectations/test_util.py:        only_for=["pandas_023"],
./tests/test_definitions/column_map_expectations/expect_column_values_to_be_of_type.json:          "only_for": ["pandas>=024"]
./tests/test_definitions/column_map_expectations/expect_column_values_to_be_of_type.json:          "only_for": ["pandas_022", "pandas_023"]
./tests/test_definitions/test_expectations.py:                                        "pandas_022" in only_for
./tests/test_definitions/test_expectations.py:                                        or "pandas_023" in only_for
./tests/test_definitions/test_expectations.py:                                if ("pandas>=024" in only_for) and (
./tests/test_definitions/test_expectations_cfe.py:                                        "pandas_022" in only_for
./tests/test_definitions/test_expectations_cfe.py:                                        or "pandas_023" in only_for
./tests/test_definitions/test_expectations_cfe.py:                                if ("pandas>=024" in only_for) and (
./docs/changelog.md:* [BUGFIX] Check for pandas>=024 not pandas>=24 (#4263)
./docs/changelog.md:* [BUGFIX] Check for pandas>=024 not pandas>=24 (#4263)
./docs_rtd/changelog.rst:* [BUGFIX] Check for pandas>=024 not pandas>=24 (#4263)
./great_expectations/self_check/util.py:    #   - only_for can be any of: pandas, pandas_022, pandas_023, pandas>=024
./great_expectations/self_check/util.py:                    "pandas_022" in expectation_test_case.only_for
./great_expectations/self_check/util.py:                    or "pandas_023" in expectation_test_case.only_for
./great_expectations/self_check/util.py:                elif "pandas>=024" in expectation_test_case.only_for:
```

Reply:
Good point on the usage stats @kenwade4 - they are currently merged into develop. It will take some time though to collect data and for @Super-Tanner to wade through and analyze.
I made a comment elsewhere, but maybe to separate the two deprecations (python 3.6 and pandas 0.23) we modify the stage to `Python37-Pandas023` instead of deleting and then EOL pandas in a separate effort.

Reply:
that sounds good! will revert these changes.

Reply:
OK, since `Python37-Pandas023` was added to the matrix, ignore the message about cleaning up special test logic for old versions of pandas